packages/iota-names/sources/admin.move

```
@@ -52,3 +52,18 @@ entry fun reserve_domains(
         iota::transfer::public_transfer(nft, sender);
     };
 }
+
+/// Admin function to forcefully remove registry entries by their domain.
+/// This bypasses all expiration and authorization checks and immediately 
+/// removes the records and invalidates any reverse lookup entries.
+public fun admin_remove_records(
+    _: &AdminCap,
+    iota_names: &mut IotaNames,
+    mut domains: vector<String>,
+) {
+    let registry = iota_names::auth_registry_mut<AdminAuth, Registry>(AdminAuth {}, iota_names);
+    while (!domains.is_empty()) {
```

Should we throw an error if the domains list is empty?

---

Changed in https://github.com/iotaledger/iota-names/pull/335/commits/e35fd734b2ca950c4d9a25cf3c3f70c3e58a901e

---

You've added it to another function?

---

Oh right, fixed